src/Nethermind/Nethermind.Merge.Plugin/Handlers/GetPayloadBodiesByHashV1Handler.cs

```diff
@@ -22,16 +23,18 @@ public GetPayloadBodiesByHashV1Handler(IBlockTree blockTree, ILogManager logMana
         _logger = logManager.GetClassLogger();
     }
 
-    public Task<ResultWrapper<ExecutionPayloadBodyV1Result?[]>> HandleAsync(Keccak[] blockHashes)
+    public Task<ResultWrapper<IEnumerable<ExecutionPayloadBodyV1Result?>>> HandleAsync(IList<Keccak> blockHashes)
     {
```

Comment:
Tbh I preferred array instead of ` IList<Keccak>` here. Not seeing the value of this interface

Reply:
You generally use the most generic interface until you have some specific needs.

Reply:
`IReadOnlyList<>` are great for being generic, but still indexable and we can use `for` on them.